tests/Aspire.Hosting.Azure.Tests/AzureEnvironmentResourceTests.cs

```diff
@@ -183,6 +183,49 @@ private sealed class TestProject : IProjectMetadata
         public LaunchSettings? LaunchSettings { get; set; }
     }
 
+    [Fact]
+    public async Task AzurePublishingContext_IgnoresAzureBicepResourcesWithIgnoreAnnotation()
+    {
+        // Arrange
+        var tempDir = Directory.CreateTempSubdirectory(".azure-ignore-annotation-test");
+        output.WriteLine($"Temp directory: {tempDir.FullName}");
+        using var builder = TestDistributedApplicationBuilder.Create(DistributedApplicationOperation.Publish,
+            publisher: "default",
+            outputPath: tempDir.FullName);
+
+        // Add Azure container app environment to ensure we have a publishing context
+        builder.AddAzureContainerAppEnvironment("acaEnv");
```

Comment:
This isn't required

Reply:
This turned out to be a bigger change. We needed to add another "exclude" check to AzureResourcePreparer. Since we duplicate this check in so many places, I introduced a new API to encapsulate the logic.